src/common/com/intellij/plugins/haxe/ide/refactoring/introduce/HaxeIntroduceHandler.java

```diff
@@ -365,17 +378,25 @@ protected void performIntroduceWithDialog(HaxeIntroduceOperation operation) {
 
   protected void performInplaceIntroduce(HaxeIntroduceOperation operation) {
     final PsiElement statement = performRefactoring(operation);
-    final HaxeComponent target = PsiTreeUtil.findChildOfType(statement, HaxeComponent.class);
+    final HaxeComponent target = statement instanceof HaxeComponent ? (HaxeComponent)statement : PsiTreeUtil.findChildOfType(statement, HaxeComponent.class);
     if (target == null) {
       return;
```

Comment:
Since PsiTreeUtil.findChildOfType never looks at itself, we would end up not finding our added statement in many cases and we would leave the job half-finished (e.g. `var ` and an error block.)